.github/workflows/lint.yml

```diff
@@ -89,9 +89,9 @@ jobs:
   cpp:
     name: Lint files with clang-format
     runs-on: ubuntu-20.04
-    # Use ubuntu:hirsute explicitly as it is the first version to have
-    # clang-format 12 available
-    container: ubuntu:hirsute
+    # Use ubuntu:jammy explicitly as it is the first version to have
+    # clang-format 14 available
+    container: ubuntu:jammy
```

Comment:
after this change, i have
```
Run find . -regex '.*\.\(cpp\|h\|hpp\|cc\|proto\|java\)' | xargs -n1 clang-format -i -style=file -fallback-style=none
[5](https://github.com/redpanda-data/redpanda/runs/6589609610?check_suite_focus=true#step:5:6)
warning: Not a git repository. Use --no-index to compare two paths outside a working tree
[6](https://github.com/redpanda-data/redpanda/runs/6589609610?check_suite_focus=true#step:5:7)
usage: git diff --no-index [<options>] <path> <path>
```

in https://github.com/redpanda-data/redpanda/runs/6591653680?check_suite_focus=true . in other words, it turns out that the `.git` directory is missing after `checkout` action completes. but the log message looks sane

> /usr/bin/git log -1 --format='%H'
> [480](https://github.com/redpanda-data/redpanda/runs/6591653680?check_suite_focus=true#step:4:488)
> 'c8efcf346500bd5209ce528e75173249300d4b23'


i tried

1. use `ubuntu-latest` as the `runs-on` node + `actions/checkout@v2`
2. use `ubuntu-20.04` as the `runs-on` node + `actions/checkout@v2`
3. use `ubuntu-20.04` as the `runs-on` node + `actions/checkout@v3`

will try harder tomorrow. before taming the jammy container and git, i'd like keep this changeset as "draft".

Reply:
turns out the git on ubuntu:jammy is not able to read the git repo cloned using the checkout action, so, in this change, i am installing clang-format packaged by LLVM instead.

Reply:
lint is green: https://github.com/tchaikov/redpanda/runs/6613169747?check_suite_focus=true